### PR TITLE
EPL-1338: Add "How to Create a Module"

### PR DESCRIPTION
**Pull request by Gremiger**

### CODE REVIEW

File: .vscode/settings.json
Comment (isaiasb-etendo) [inline .vscode/settings.json:1]: ?

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+are marked as _In Development_ can be used in the development process. A
+module can consist of the following software artifacts:
+
+  * Application dictionary components: all new application dictionary artifacts must be associated with a module. Application dictionary windows have a field that enables you to link an application dictionary component to a module. 
+  * Software resources: these are artifacts not defined in Etendo Classic itself but, for example, Java classes, libraries, web resources (images, etc.), configuration files, etc. To maintain the relationship with the module, software resources need to be located in the module's directory inside the Etendo Classic development project. Each module has its own unique directory. 
+  * Reference data: modules can be distributed with their own set of reference business data, for example, product categories, tax code, etc. The reference data is defined in datasets which can be related to a module. 
+
+The process of developing a module has three main steps:
+
+  1. Register your module in the Application Dictionary and in the central repository.
```
Comment (isaiasb-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:55]: The concept of central repository no longer exists, customization modules are published by partners in their own repositories.

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  * Software resources: these are artifacts not defined in Etendo Classic itself but, for example, Java classes, libraries, web resources (images, etc.), configuration files, etc. To maintain the relationship with the module, software resources need to be located in the module's directory inside the Etendo Classic development project. Each module has its own unique directory. 
+  * Reference data: modules can be distributed with their own set of reference business data, for example, product categories, tax code, etc. The reference data is defined in datasets which can be related to a module. 
+
+The process of developing a module has three main steps:
+
+  1. Register your module in the Application Dictionary and in the central repository. 
+  2. Develop the artifacts included in your module. Depending on the functional specification and technical design of your module, it might include only one type of artifacts or a combination of them. In following sections, each type of artifact is described in detail. 
+  3. Publish the module in the central repository.
```
Comment (isaiasb-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:57]: Same

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    modules, including customizations. You can make changes directly in
+    other modules -including Etendo Classic core- but it is highly recommended not
+    to do that. It makes it much easier to maintain Etendo Classic if you restrict
+    code changes to modules.
+
+A module can be distributed and downloaded by other Etendo Classic users via
+the central repository. For more information about the central repository and
+distribution of modules, visit the [Modularity Guide](/concepts/Modularity_Concepts.md).
+
+The following sections discuss the main topic of this section: create and
```
Comment (isaiasb-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:71]: Link to https://docs.etendo.software/latest/developer-guide/etendo-classic/how-to-guides/how-to-publish-modules-to-github-repository/

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    For a detailed description of this `export.database` task and other relevant
+    Module related gradle tasks, see the [database gradle tasks](concepts/Development_Build_Tasks#database_tasks) and [module gradle task](concepts/Development_Build_Tasks#Modules) descriptions.
+
+
+##  Creating a Source Directory
+
+To develop manual Java code you need a _src_ directory inside of your specific
+module:
+
+![](/assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-5.png)
```
Comment (isaiasb-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:191]: same

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+For detailed information about this process, visit the [How to Publish Modules to Github Repository guide](how-to-publish-modules-to-github-repository.md)
+
+##  The Result
+
+The result of this section is a correctly set up module, which can be installed in an Etendo Classic instance that meets the module's requirements.
+
+---
+
+This work is a derivative of [How to Create and Package a Module](http://wiki.Etendo.com/wiki/How_To_Create_and_Package_a_Module){target="\_blank"} by [Openbravo Wiki](http://wiki.openbravo.com/wiki/Welcome_to_Openbravo){target="\_blank"}, used under [CC BY-SA 2.5 ES](https://creativecommons.org/licenses/by-sa/2.5/es/){target="\_blank"}. This work is licensed under [CC BY-SA 2.5](https://creativecommons.org/licenses/by-sa/2.5/){target="\_blank"} by [Etendo](https://etendo.software){target="\_blank"}.
```
Comment (isaiasb-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:209]: http://wiki.openbravo.com/wiki/How_To_Create_and_Package_a_Module

**Review by isaiasb-etendo — CHANGES_REQUESTED** [review]

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md (hunk 3 above)
Comment (Gremiger) [reply docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:71]: Added link

**Review by isaiasb-etendo — APPROVED** [review]

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  * To provide the community with a rich set of extensions to meet their unique business requirements without bloating the core product. 
+  * To shorten the implementation cycles by enabling system integrators to develop very specific functionality to meet the unique needs of a particular market. 
+
+A module is a piece of additional functionality that can be deployed
+optionally and independently on top of Etendo Classic. Examples of modules are:
+additional reports, additional windows, connectors, content packs
+(translations, chart of accounts, list of tax codes, product categories, etc).
+
+!!!info
+    For a detailed description on the Etendo Classic modularity concept, see the
```
Comment (RomanMagnoli) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:39]: Delete the unnecessary line breaker.

Check the rest of them, like in line 42

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+  * Software resources: these are artifacts not defined in Etendo Classic itself but, for example, Java classes, libraries, web resources (images, etc.), configuration files, etc. To maintain the relationship with the module, software resources need to be located in the module's directory inside the Etendo Classic development project. Each module has its own unique directory. 
+  * Reference data: modules can be distributed with their own set of reference business data, for example, product categories, tax code, etc. The reference data is defined in datasets which can be related to a module. 
+
+The process of developing a module has three main steps:
+
+  1. Register your module in the Application Dictionary and in your github repository.
+  2. Develop the artifacts included in your module. Depending on the functional specification and technical design of your module, it might include only one type of artifacts or a combination of them. In following sections, each type of artifact is described in detail. 
+  3. Publish the module in your github repository.
+
+!!!Important
```
Comment (RomanMagnoli) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:59]: Does this `Important` Admonition exist? Maybe I'm wrong, but the comment is based on: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types

Comment (Gremiger) [reply docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:59]: Yep, it does exists

image

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+A module can be distributed and downloaded by other Etendo Classic users via the central repository. For more information about the central repository and distribution of modules, visit the [Modularity Guide](../concepts/Modularity_Concepts.md).
+
+The following sections discuss the main topic of this section: create and setup a module, and [publish it for distribution](how-to-publish-modules-to-github-repository.md).
+
+##  Creating a Module
+
+The first step in the development process is to create a new module. From the Application menu, select `Application Dictionary` > `Module`.
+
+![](/assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-0.png)
```
Comment (valeg-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:57]:
```suggestion
![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-0.png)
```

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+In this section, we create an example module, called _Etendo Howtos_ .
+We will define the module using the following three tabs of the `Application Dictionary` > `Module` window:
+
+  * Dependency
+  * DB Prefix
+  * Data Package
+
+Dependency: Etendo Classic core functionality is in itself a module, called _Core_. All modules have a dependency on User Interface Application and in turn, it has a dependency on Etendo core. In the Dependency window, specify that the module is dependent on User Interface Application. (see also the  AD_Module_Dependency description):
+
+![](/assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-1.png)
```
Comment (valeg-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:87]:
```suggestion
![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-1.png)
```

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+
+  
+DB Prefix: at least one db prefix is required when a module also contains database artifacts (table, column, constraints, etc.). Etendo Classic determines the module of a database artifact by checking to see if its name starts with one of the defined db_prefixes. If database artifacts will be added to the module, at least one DB prefix must be specified.
+
+!!!important
+    DB Prefix can only contain uppercase [A-Z0-9] and the first letter is required to be from [A-Z] only.  
+
+  
+![](/assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-2.png)
```
Comment (valeg-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:96]:
```suggestion
![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-2.png)
```

File: docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+
+!!!info
+    For more information, see the  AD_Module_Dbprefix  table description.
+
+  
+Data Package: Tables are linked to a module through the Data Package. The generated business object for the new table will use the Java Package defined in the  Data Package. If new tables are to be added to the module, a data package must be specified as illustrated in the image below.
+
+![](/assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-3.png)
```
Comment (valeg-etendo) [inline docs/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module.md:104]:
```suggestion
![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_To_Create_a_Module-3.png)
```

**Review by valeg-etendo — CHANGES_REQUESTED** [review]

**Review by valeg-etendo — APPROVED** [review]

**Pull request merged**